src/queue.c

```diff
@@ -5776,11 +5782,13 @@ _dispatch_queue_cleanup2(void)
 	// overload the "probably" variable to mean that dispatch_main() or
 	// similar non-POSIX API was called
 	// this has to run before the DISPATCH_COCOA_COMPAT below
+#ifndef __linux__
```

Comment:
and here I would comment that Linux parks the main thread into sigsuspend hence this code is useless for this platform